versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5846 @@
+
+**Returns**: An `Object`, with as many members as there are unique names in the TSV.
+
+**Example**
+
+This task writes a TSV file to standard out, then reads it into an `Object`.
+
+```wdl
+task test {
+  command <<<
+    python <<CODE
+    print('\t'.join(["key_{}".format(i) for i in range(3)]))
+    print('\t'.join(["value_{}".format(i) for i in range(3)]))
+    CODE
+  >>>
+  output {
+    Object my_obj = read_object(stdout())
+  }
+}
+```
+
+The command outputs the following lines to stdout:
+
+```
+key_1\tkey_2\tkey_3
+value_1\tvalue_2\tvalue_3
+```
+
+Which are read into an `Object` with the following members:
+
+|Attribute|Value|
+|---------|-----|
+|key_1    |"value_1"|
+|key_2    |"value_2"|
+|key_3    |"value_3"|
+
+## 🗑 Array[Object] read_objects(String|File)
+
+Reads a tab-separated value (TSV) file representing the names and values of the members of any number of `Object`s. 
+
+There must be a header row with the names of the object members. The names in the first row must be unique; if there are any duplicate names, an error is raised.
+
+There are any number of additional rows, where each additional row contains the values of an object corresponding to the member names. Every row in the file must have the same number of elements.
+
+If the entire contents of the file can not be read for any reason, the calling task or workflow fails with an error. Examples of failure include, but are not limited to, not having access to the file, resource limitations (e.g. memory) when reading the file, and implementation-imposed file size limits.
+
+**Parameters**
+
+1. `String|File`: Path of the TSV file to read. If the argument is a `String`, it is assumed to be a local file path relative to the current working directory of the task.
+
+**Returns**: An `Array[Object]`, with N-1 elements, where N is the number of rows in the file.
+
+**Example**
+
+This task writes a TSV file to standard out, then reads it into an `Array[Object]`.
+
+```wdl
+task test {
+  command <<<
+    python <<CODE
+    print('\t'.join(["key_{}".format(i) for i in range(3)]))
+    print('\t'.join(["value_{}".format(i) for i in range(3)]))
+    print('\t'.join(["value_{}".format(i) for i in range(3)]))
+    print('\t'.join(["value_{}".format(i) for i in range(3)]))
+    CODE
+  >>>
+  output {
+    Array[Object] my_obj = read_objects(stdout())
+  }
+}
+```
+
+The command outputs the following lines to stdout:
+
+```
+key_1\tkey_2\tkey_3
+value_1\tvalue_2\tvalue_3
+value_1\tvalue_2\tvalue_3
+value_1\tvalue_2\tvalue_3
+```
+
+Which are read into an `Array[Object]` with the following elements:
+
+|Index|Attribute|Value|
+|-----|---------|-----|
+|0    |key_1    |"value_1"|
+|     |key_2    |"value_2"|
+|     |key_3    |"value_3"|
+|1    |key_1    |"value_1"|
+|     |key_2    |"value_2"|
+|     |key_3    |"value_3"|
+|2    |key_1    |"value_1"|
+|     |key_2    |"value_2"|
+|     |key_3    |"value_3"|
+
+## X read_json(String|File)
```

Comment:
Sounds like this:
## X read_json(String|File)

should actually be: 
## R read_json(String|File)

as per the other comment I left - you mention arbitrary Types as `R` in your standard library intro, so shouldn't use `X` here as the type.

Reply:
yep, you're right here